nilearn/decomposition/canica.py

```diff
@@ -127,13 +127,18 @@ def __init__(self, mask=None, n_components=20, smoothing_fwhm=6,
             mask_strategy=mask_strategy, mask_args=mask_args,
             memory=memory, memory_level=memory_level,
             n_jobs=n_jobs, verbose=verbose)
+
+        if(threshold > n_components):
```

Comment:
tests are failing because of this change. You have any issue or idea behind this change ? Because I could not understand it properly.